CIScripts/TestRunner/Invoke-PesterTests.ps1

```diff
@@ -1,39 +1,26 @@
 function Invoke-PesterTests {
     Param (
         [Parameter(Mandatory = $true)] [String] $TestRootDir,
-        [Parameter(Mandatory = $false)] [String] $ReportDir,
+        [Parameter(Mandatory = $false)] [String] $ReportPath,
         [Parameter(Mandatory = $false)] [String[]] $IncludeTags,
         [Parameter(Mandatory = $false)] [String[]] $ExcludeTags,
-        [Parameter(Mandatory = $false)] [System.Collections.Hashtable] $AdditionalParams
+        [Parameter(Mandatory = $false)] [System.Collections.Hashtable] $AdditionalParams,
+        [Parameter(Mandatory = $false)] [String[]] $CodeCovFiles = @()
     )
-    if ($ReportDir) {
-        if (-not (Test-Path $ReportDir)) {
-            New-Item -ItemType Directory -Path $ReportDir | Out-Null
-        }
-    }
 
-    $TestPaths = Get-ChildItem -Path $TestRootDir -Recurse -Filter "*.Tests.ps1"
-    $TotalResults = @{
-        PassedCount = 0;
-        FailedCount = 0;
+    $PesterScript = @{
+        Path=$TestRootDir;
+        Parameters=$AdditionalParams;
+        Arguments=@();
     }
-    foreach ($TestPath in $TestPaths) {
-        $PesterScript = @{
-            Path=$TestPath.FullName;
-            Parameters=$AdditionalParams;
-            Arguments=@();
-        }
-        $Basename = $TestPath.Basename
-        if ($ReportDir) {
-            $TestReportOutputPath = "$ReportDir\$Basename.xml"
-            $Results = Invoke-Pester -PassThru -Script $PesterScript -Tags $IncludeTags -ExcludeTag $ExcludeTags `
-                -OutputFormat NUnitXml -OutputFile $TestReportOutputPath
-        } else {
-            $Results = Invoke-Pester -PassThru -Script $PesterScript -Tags $IncludeTags -ExcludeTag $ExcludeTags
-        }
-
-        $TotalResults.PassedCount += $Results.PassedCount
-        $TotalResults.FailedCount += $Results.FailedCount
+    if ($ReportPath) {
```

Comment:
While it's true that `$ReportPath` is always passed, sometimes it's a nil object. 

Notice, that `./Invoke-Selfcheck.ps1` specifies that `$ReportPath` doesn't have to be mandatory, and it is still passed to `Invoke-PesterTests`. The check happens in `Invoke-PesterTests` to decide on path of execution.

If it were mandatory, the following would happen:

```
PS C:\Users\mk\Juniper\tmp\contrail-windows-ci> function foo {
>>   Param ([Parameter(Mandatory=$true)] [string] $x)
>>   Write-Host "x is $x";
>>   if ($x) { Write-Host $x }
>> }
PS C:\Users\mk\Juniper\tmp\contrail-windows-ci> foo -x "hello"
x is hello
hello
PS C:\Users\mk\Juniper\tmp\contrail-windows-ci> foo

cmdlet foo at command pipeline position 1
Supply values for the following parameters:
x:
foo : Cannot bind argument to parameter 'x' because it is an empty string.
At line:1 char:1
+ foo
+ ~~~
    + CategoryInfo          : InvalidData: (:) [foo], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,foo
```

So, I'm not sure about this.

However, I'm pretty sure we don't want to output the report file every time. This should remain optional imho.

Reply:
Ok, sounds good.